EngineLayer/Proteomics/CompactPeptide.cs

```diff
@@ -44,6 +44,13 @@ public CompactPeptide(PeptideWithSetModifications peptideWithSetModifications)
             MonoisotopicMassIncludingFixedMods = peptideWithSetModifications.MonoisotopicMass;
         }
 
+        public CompactPeptide(CompactPeptide cp, double MonoisotopicMassIncludingFixedMods)
+        {
+            this.CTerminalMasses = cp.CTerminalMasses;
+            this.NTerminalMasses = cp.NTerminalMasses;
+            this.MonoisotopicMassIncludingFixedMods = MonoisotopicMassIncludingFixedMods;
+        }
```

Comment:
It's to generate a new CompactPeptide with an updated precursor mass. This was intended.